### PR TITLE
Initial support for openSUSE Leap + Tumbleweed

### PR DESCRIPTION
**Pull request by Jamdoog**
- [x] Adds `install.yml` for OpenSUSE Leap and Tumbleweed
- [x] Adds `uninstall.yml` for OpenSUSE Leap and Tumbleweed

### CODE REVIEW

**Comment by Jamdoog** [comment]
Hello,

This is my first attempt at contributing to OSS outside of my own repositories. Please let me know if I have done something wrong. 


I have tried to add support for openSUSE Leap 15.1-15.3 and Tumbleweed to this role. I have been unable to test the role fully due to systemd issues in containers on my device. However, I did snoop around and could see that this installed successfully:

`:/ # cat /usr/lib/systemd/system/tailscaled.service 
[Unit]
Description=Tailscale node agent
...
WantedBy=multi-user.target`

`:/ # tailscale
USAGE
  tailscale [flags] <subcommand> [command flags]
..
FLAGS
  --socket string
    	path to tailscaled's unix socket (default /var/run/tailscale/tailscaled.sock)`

I have tried multiple SUSE images, @mesaguy 's image has given me success previously but I can't get it working here.
(https://hub.docker.com/r/mesaguy/ansible-molecule-opensuse/tags)

Please let me know how I can improve my pull request for it to be accepted in the main branch.

Thanks.

File: meta/main.yml [hunk 1]
```diff
@@ -38,6 +38,11 @@ galaxy_info:
     - name: ArchLinux
       versions:
         - all
+    - name: opensuse
+      versions:
+        - 15.1
+        - 15.2
+        - 15.3
```
Comment (artis3n) [inline meta/main.yml:45]: These need to be strings to satisfy ansible-lint + galaxy's schema expectations.

File: tasks/opensuse/install.yml [hunk 2]
```diff
@@ -0,0 +1,18 @@
+---
+- name: openSUE | Install Repo
```
Comment (artis3n) [inline tasks/opensuse/install.yml:2]:
```suggestion
- name: OpenSUSE | Install Repo
```

Update these to satisfy the linting case expectation

File: vars/main.yml [hunk 3]
```diff
@@ -14,6 +14,15 @@ centos_family_distros:
   - Rocky
   - AlmaLinux
 
+opensuse_family_distros:
+  - "openSUSE Leap"
+  - "openSUSE Tumbleweed"
+
+opensuse_leap_family_supported_releases:
```
Comment (artis3n) [inline vars/main.yml:21]: What is this variable used for?

File: tasks/opensuse/install.yml [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+  community.general.zypper_repository:
```
Comment (artis3n) [inline tasks/opensuse/install.yml:3]: These tasks need `become: true` for sudo

File: tasks/opensuse/install.yml [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+    repo: "{{ opensuse_tumbleweed_repository }}"
+    state: present
+  when: ansible_distribution == "openSUSE Tumbleweed"
+
+- name: openSUSE | Install Repo
+  community.general.zypper_repository:
+    repo: "{{ opensuse_leap_repository }}"
+    state: present
```
Comment (artis3n) [inline tasks/opensuse/install.yml:11]: Make these `state: '{{ state }}'` so they are cleaned up as well if users wish to uninstall

**Review by artis3n — COMMENTED** [review]
Thank you for this PR! I've been delayed getting to these distros by needing to create a container I can run through the Molecule unit tests. I'll take a look at the container you mention and see if I can get it working or build off of it to make a working version.

In the meantime, added some minor change requests. The failing Debian 12 test is unrelated and due to https://github.com/artis3n/ansible-role-tailscale/issues/257.

File: tasks/opensuse/install.yml (hunk 5 above)
Comment (Jamdoog) [reply tasks/opensuse/install.yml]: https://github.com/Jamdoog/ansible-role-tailscale/commit/8a4d2d20087d7abed2c4cb8700a38a4b5c2e0147

File: tasks/opensuse/install.yml (hunk 4 above)
Comment (Jamdoog) [reply tasks/opensuse/install.yml:4]: https://github.com/Jamdoog/ansible-role-tailscale/commit/da0ac7a035b4a76a84e69867ef23ef7f2df0bd1b

File: vars/main.yml (hunk 3 above)
Comment (Jamdoog) [reply vars/main.yml:17]: https://github.com/Jamdoog/ansible-role-tailscale/commit/7b6aeda0d080d076cbf7a1a3e4c2def56deb2c33

File: tasks/opensuse/install.yml (hunk 2 above)
Comment (Jamdoog) [reply tasks/opensuse/install.yml]: https://github.com/Jamdoog/ansible-role-tailscale/commit/ab9305f3eeba8dc3a02e64f3867141ecc431523a

File: meta/main.yml (hunk 1 above)
Comment (Jamdoog) [reply meta/main.yml]: https://github.com/Jamdoog/ansible-role-tailscale/commit/aae01e5177d36fb7c591d76248da5363c8ac8116

**Comment by Jamdoog** [comment]
> Thank you for this PR! I've been delayed getting to these distros by needing to create a container I can run through the Molecule unit tests. I'll take a look at the container you mention and see if I can get it working or build off of it to make a working version.
> 
> In the meantime, added some minor change requests. The failing Debian 12 test is unrelated and due to #257.

Hi, I've done the changes you asked. Looks like it's applied here :)

Let me know if I can help anymore. 

Thanks

File: vars/main.yml (hunk 3 above)
Comment (artis3n) [reply vars/main.yml:17]: I think you removed the wrong variable!

Comment (Jamdoog) [reply vars/main.yml]: Hi, hopefully this is resolved now (?)

File: tasks/opensuse/install.yml [hunk 6]
```diff
@@ -0,0 +1,21 @@
+---
+- name: OpenSUE | Install Repo
+  become: true
+  community.general.zypper_repository:
+    repo: "{{ opensuse_tumbleweed_repository }}"
+    state: '{{ state }}'
```
Comment (artis3n) [inline tasks/opensuse/install.yml:6]: Woops I was wrong, forgot how I have the rest of the distros set up. These should be `state: present`, and we'll want to add an `uninstall.yml` that does the opposite.

File: tasks/opensuse/install.yml [hunk 7]
```diff
@@ -0,0 +1,21 @@
+---
+- name: OpenSUE | Install Repo
```
Comment (artis3n) [inline tasks/opensuse/install.yml:2]: To make the output more readable:

```suggestion
- name: OpenSUE | Install Repo - Tumbleweed
```

File: tasks/opensuse/install.yml [hunk 8]
```diff
@@ -0,0 +1,21 @@
...
+  become: true
+  community.general.zypper_repository:
+    repo: "{{ opensuse_tumbleweed_repository }}"
+    state: '{{ state }}'
+  when: ansible_distribution == "openSUSE Tumbleweed"
+
+- name: OpenSUSE | Install Repo
```
Comment (artis3n) [inline tasks/opensuse/install.yml:9]: To make the output more readable:

```suggestion
- name: OpenSUSE | Install Repo - Leap
```

**Review by artis3n — COMMENTED** [review]
~I have also failed to get systemd working in an opensuse container the last few days.~ The opensuse tasks appear to work until the systemd steps, though, so I think we can merge this and work on getting a reproducible test environment up for the repo later.

To be ready to merge, please add an `opensuse/uninstall.yml` similar to the `install.yml` - with state pinned to `absent`.

EDIT: I have working opensuse containers for us to add into the CI tests! See comment below.

**Comment by artis3n** [comment]
1) If you merge the latest changes from `main`, the failing Debian 12 test will be fixed again.

2) I have working opensuse containers and verified this PR can successfully deploy Tailscale to them through the molecule suite. Given the way the `pull_request_target` trigger works, it won't do anything to add it to your current PR, and adding it before we have tasks set to handle them won't be pretty either. So I'll add them in a follow-up PR and no action needed from you.

**Review by artis3n — APPROVED** [review]
I appreciate your patience!

**Comment by Jamdoog** [comment]
> I appreciate your patience!

No worries.

It says that my merge is blocked due to it not being signed. Is this still true with manual approval?

**Comment by artis3n** [comment]
I'm going to squash merge this in which case GitHub should sign it, so it should be fine. But that is a reminder to write up that DEVELOPMENT.md I've been meaning to do.

**Comment by artis3n** [comment]
I've never seen this error that's currently hitting the tests, it makes me think it's a GitHub error or outage of some kind. I'm going to wait a bit and rerun tests to see if it resolves, else I'll just merge this and figure it out later. This works against the opensuse containers I tested.

**Comment by artis3n** [comment]
Yup, all the images run fine on this PR, including the new openSUSE ones. Something broke in the PR environment, but I'm overriding to merge.

**Comment by artis3n** [comment]
The CI errors were due to unlucky timing with `ubuntu-latest` rolling over to ubuntu-22.04. Talking to systemd from a Docker container via Molecule doesn't seem to work

https://github.com/artis3n/ansible-role-tailscale/issues/266

**Pull request merged**